### ISSUE TITLE
Modular Driver JAR for Java9

### ISSUE THREAD

**Issue by thomasklingler** [Microsoft/mssql-jdbc#569]
## Driver version or jar name
mssql-jdbc-6.3.5.jre8-preview.jar

## Java/JVM version
JDK 9.0.1

## Proposal
Provide a modular **mssql-jdbc-mod-6.3.5-preview.jar** for use with Java 9.
Add the following module-info:
```
import com.microsoft.sqlserver.jdbc.SQLServerDriver;
module mssql.jdbc {
	provides java.sql.Driver with SQLServerDriver;
	requires java.logging;
	requires java.naming;
	requires java.security.jgss;
	requires java.sql;
	requires java.xml;
	requires java.xml.bind;
}
```
The module-info could be extended to make additional public types available for direct reference:
```
exports com.microsoft.sqlserver.jdbc;
```
I tried this successfully on a copy of mssql-jdbc-6.3.5.jre8-preview.jar by adding a compiled `module-info.class` file, and removing the now invalid signature files from the META-INF directory.
Note that the modified jar still works for Java8: Java8 ignores the module-info class, and all other classes are compiled with Java8.

**Comment by rene-ye** [comment]
Hi @thomasklingler, the push request for an updated JDBC4.3 with module-info is currently undergoing review. In the mean time, you can test out the changes from [this branch](https://github.com/rene-ye/mssql-jdbc/tree/JDBC4.3). Please feel free to provide feedback on the changes, and let me know if something doesn't work or can be improved.

**Comment by softboy99** [comment]
waiting for good news

**Comment by cheenamalhotra** [comment]
Hi @softboy99 

We added 'automatic-module-name' to the Manifest such that the `jre10` driver Jar can now be included in modulePath and module `com.microsoft.sqlserver.jdbc` will be available for use in Java 10 applications. The change has been merged with PR #732 and will be available in next driver release.
